conf/svcomp21.json

```diff
@@ -28,6 +28,7 @@
     "partition-arrays": {
       "enabled": true
     },
+    "lower-constants" : false,
```

Comment:
This should only be enabled when checking for overflows, otherwise it is helpful

Reply:
Instead of putting it into svcomp21 you could toggle the option here:

https://github.com/goblint/analyzer/blob/03c5da2f4fd913fd242ae48e10a96acda4f64ee9/src/witness/witness.ml#L277-L287

It might make more sense to fix the unsoundness in CIL itself, then we wouldn't need such toggle.

Reply:
> Instead of putting it into svcomp21 you could toggle the option here:
I'll try that, it might be too late though, this needs to be set on the call to `initCil()` already.

>It might make more sense to fix the unsoundness in CIL itself, then we wouldn't need such toggle.

It is not really a unsoundness, CIL just assumes that the programs have no undefined behavior. This is once again this issue that what it means to be "sound" for programs with UB is hard to say, as these programs do not have any meaningful semantics from the C perspective.

This option is very helpful, because it allows, e.g. 
```
enum
{
  _ISupper = ((0) < 8 ? ((1 << (0)) << 8) : ((1 << (0)) >> 8)),
  _ISlower = ((1) < 8 ? ((1 << (1)) << 8) : ((1 << (1)) >> 8))
};
```

which, when this option is disabled will fail as CIL tries to generate `if` `else` statements which can not appear at a global scope (this actually causes us to fail with an exception on some of the `BusyBox-NoOverflow` tests when this toggle is on).

So, CIL itself would need to be able to check if a constant expression might overflow and in this case not do the lowering, but keep doing it otherwise. I don't think that it is realistic to get this working for sv-comp.

Reply:
> I'll try that, it might be too late though, this needs to be set on the call to `initCil()` already.

It's being set after `initCIL ()` though already.

Reply:
True, definitely before CIL parses the file though.
I'll just try and see if works there, otherwise I'll find a different sport for it.

Reply:
Ok, we can indeed set it here.
The other question is if we don't want different config files for different properties in general? This would probably be cleaner. Do you know if it is allowed by the rules to have different configs per property?

Reply:
I set this from `Maingoblint.main` now.